### PR TITLE
hwcontext_qsv: add a null check to avoid dereferencing null pointer

### PR DESCRIPTION
**Pull request by tong1wu**
When the new_hdlpair = NULL, it could cause a dereferencing null pointer
error. Now add a null check to avoid this possible error.

Signed-off-by: Tong Wu <tong1.wu@intel.com>

### CODE REVIEW

**Comment by tong1wu** [comment]
> Please change 0082 look this order already been use

Yes, what I did is just modify patch 0082. @xhaihao Could you please help to review this patch?

File: patches/0082-libavcodec-qsvenc-Add-support-to-qsv-to-encode-exter.patch [hunk 1]
```diff
@@ -175,7 +176,7 @@ index 0d85ae7b53..5f28cb631d 100644
 +qsv_map_to_err:
 +    if (new_sur)
 +        av_freep(&new_sur);
-+    if (new_hdlpair->first)
++    if (new_hdlpair && new_hdlpair->first)
 +        av_freep(&new_hdlpair->first);
 +    if (new_hdlpair)
 +        av_freep(&new_hdlpair);
```
Comment (xhaihao) [inline patches/0082-libavcodec-qsvenc-Add-support-to-qsv-to-encode-exter.patch:182]:
```suggestion
+        av_freep(&new_hdlpair->first);
+    if (new_hdlpair)
+        av_freep(&new_hdlpair);

if (new_hdlpair) {
    if (new_hdlpair->first)
        av_freep(&new_hdlpair->first);
    av_freep(&new_hdlpair);
}
```

**Comment by tong1wu** [comment]
I will resubmit this patch later, thx.

**Comment by tong1wu** [comment]
@xhaihao I have updated the patch as suggested, please review thx.

**Review by xhaihao — APPROVED** [review]

**Pull request merged**